### PR TITLE
Factor out {user,builtin,core} plugins from plugin

### PR DESCRIPTION
**Pull request by ransombriggs**

### CODE REVIEW

**Comment by ransombriggs** [comment]
@dickeyxxx could you review?  In a PR after this I will do a refactor like yours where you moved everything to a `plugins` directory, but doing this first made it more easy to grok

File: src/builtin_plugins.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  }
+
+  plugins: Plugins
+
+  /**
+   * list user plugins
+   * @returns {Plugin[]}
+   */
+  list (): Plugin[] {
+    let commandsPath = path.resolve(path.join(__dirname, 'commands'))
```
Comment (jdx) [inline src/builtin_plugins.js]: I'd prefer this to be in a property I think

Comment (jdx) [reply src/builtin_plugins.js]: also, any particular reason for using `path.resolve` here? My understanding is it is the same unless you're using relative paths, so it should be fine

File: src/builtin_plugins.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+export default class BuiltinPlugins {
+  constructor (plugins: Plugins) {
+    this.plugins = plugins
+  }
+
+  plugins: Plugins
+
+  /**
+   * list user plugins
```
Comment (jdx) [inline src/builtin_plugins.js]: `list builtin plugins`

File: src/builtin_plugins.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+   * @returns {Plugin[]}
+   */
+  list (): Plugin[] {
+    let commandsPath = path.resolve(path.join(__dirname, 'commands'))
+    return [new Plugin('builtin', commandsPath, this.plugins)]
```
Comment (jdx) [inline src/builtin_plugins.js]: should certainly go in a different PR, but thoughts on making this an options instead?

```js
return [new Plugin({type: 'builtin', path: commandsPath, plugins: this.plugins})]
```

**Review by jdx — CHANGES_REQUESTED** [review]

File: src/builtin_plugins.js (hunk 2 above)
Comment (ransombriggs) [reply src/builtin_plugins.js]: fixed in 89f7f8acc2be74174709cc4632b8458f6ad07748

File: src/builtin_plugins.js (hunk 1 above)
Comment (ransombriggs) [reply src/builtin_plugins.js]: fixed in 89f7f8acc2be74174709cc4632b8458f6ad07748

File: src/builtin_plugins.js (hunk 3 above)
Comment (ransombriggs) [reply src/builtin_plugins.js]: I will put in a different PR after I refactor `Plugin`

**Review by jdx — APPROVED** [review]
